test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
+        {
+            buf = 'Aug',
+            fmt = '%b',
+        },
+        {
+            buf = 'Sep',
+            fmt = '%b',
+        },
+        {
+            buf = 'Oct',
+            fmt = '%b',
+        },
+        {
+            buf = 'Nov',
+            fmt = '%b',
+        },
+        {
+            buf = 'Dec',
+            fmt = '%b',
+        },
+        -- %H
+        {
+            buf = '00',
+            fmt = '%H',
+        },
+        {
+            buf = '23',
+            fmt = '%H',
+        },
+        -- %I
+        {
+            buf = '01',
+            fmt = '%I',
+        },
+        {
+            buf = '12',
+            fmt = '%I',
+        },
+        -- %j
+        {
+            buf = '001',
+            fmt = '%j',
+        },
+        {
+            buf = '31-12-2024 366',
+            fmt = '%d-%m-%Y %j',
+        },
+        -- %k
+        {
+            buf = ' 0',
+            fmt = '%k',
+        },
+        {
+            buf = '23',
+            fmt = '%k',
+        },
+        -- %l
+        {
+            buf = ' 1',
+            fmt = '%l',
+        },
+        {
+            buf = '12',
+            fmt = '%l',
+        },
+        -- %m
+        {
+            buf = '01',
+            fmt = '%m',
+        },
+        {
+            buf = '12',
+            fmt = '%m',
+        },
+        -- %M
+        {
+            buf = '00',
+            fmt = '%M',
+        },
+        {
+            buf = '59',
+            fmt = '%M',
+        },
+        -- %n
+        {
+            buf = '\n',
+            fmt = '%n',
+        },
+        -- %O, modifier, skipped.
+        -- %p
+        {
+            buf = 'PM',
+            fmt = '%p',
+        },
+        {
+            buf = 'AM',
+            fmt = '%p',
+        },
+        -- %r, same as "%I:%M:%S %p".
+        {
+            buf = '12:32:01 AM',
+            fmt = '%r',
+        },
+        {
+            buf = '12:32:01 PM',
+            fmt = '%r',
+        },
+        -- %R, same as "%H:%M".
+        {
+            buf = '18:20',
+            fmt = '%R',
+        },
+        -- %s
+        {
+            buf = '26-08-2024 1724630400',
+            fmt = '%d-%m-%Y %s',
+        },
+        -- %S
+        {
+            buf = '00',
+            fmt = '%S',
+        },
+        {
+            buf = '59',
+            fmt = '%S',
+        },
+        -- %t
+        {
+            buf = '26-08-2024 \t',
+            fmt = '%d-%m-%Y %t',
+        },
+        -- %T, same as "%H:%M:%S".
+        {
+            buf = '00:00:00',
+            fmt = '%T',
+        },
+        {
+            buf = '23:59:59',
+            fmt = '%T',
+        },
+        -- %u
+        -- Works for 4 ('Thursday') only when date is not
+        -- complete.
+        {
+            buf = '4',
+            fmt = '%u',
+        },
+        {
+            buf = '30-08-2024 5',
+            fmt = '%d-%m-%Y %u',
+        },
+        {
+            buf = '26-08-2024 1',
```

Comment:
Why don't check all 7 days like in previous tests?
Also, don't get the sorting order.

Reply:
Fixed:

```diff
         -- complete.
         {
             buf = '4',
             fmt = '%u',
         },
         {
-            buf = '30-08-2024 5',
+            buf = '12-08-2024 1',
             fmt = '%d-%m-%Y %u',
         },
         {
-            buf = '26-08-2024 1',
+            buf = '13-08-2024 2',
+            fmt = '%d-%m-%Y %u',
+        },
+        {
+            buf = '14-08-2024 3',
+            fmt = '%d-%m-%Y %u',
+        },
+        {
+        {
+            buf = '15-08-2024 4',
+            fmt = '%d-%m-%Y %u',
+        },
+        {
+            buf = '16-08-2024 5',
+            fmt = '%d-%m-%Y %u',
+        },
+        {
+            buf = '17-08-2024 6',
             fmt = '%d-%m-%Y %u',
         },
```